internal/cmd/operator-sdk/generate/kustomize/manifests.go

```diff
@@ -174,10 +175,12 @@ func (c manifestsCmd) run(cfg *config.Config) error {
 		gencsv.WithBase(c.inputDir, c.apisDir, c.interactiveLevel),
 		gencsv.WithBaseWriter(c.outputDir),
 	}
-	if err := csvGen.Generate(cfg, opts...); err != nil {
+	/*if err := csvGen.Generate(opts...); err != nil {
+		return fmt.Errorf("error generating kustomize bases: %v", err)
+	}*/
+	if err := csvGen.GenerateWithConfig(cfg, opts...); err != nil {
 		return fmt.Errorf("error generating kustomize bases: %v", err)
 	}
```

Comment:
Then
```suggestion
	}
	csvBytes, err := yaml.Marshal(csv)
	if err != nil {
		return fmt.Errorf("error marshaling CSV base: %v", err)
	}
	if err = ioutil.WriteFile(basePath, csvBytes, 0644); err != nil {
		return fmt.Errorf("error writing CSV base: %v", err)
	}
```

Reply:
If I remove the abstraction from `generate manifests`, it will make it harder to test it.

Reply:
You can wrap these calls in a mock-able way, ex.:

```go
type csvWriteFunc func(*v1alpha1.ClusterServiceVersion, string) error

var writeCSV csvWriteFunc = writeCSVFile

func writeCSVFile(csv *v1alpha1.ClusterServiceVersion, path string) error {
	csvBytes, err := yaml.Marshal(csv)
	if err != nil {
		return fmt.Errorf("error marshaling CSV base: %v", err)
	}
	if err = ioutil.WriteFile(path, csvBytes, 0644); err != nil {
		return fmt.Errorf("error writing CSV base: %v", err)
	}
	return nil
}

func (c manifestsCmd) run(cfg *config.Config) error {
	...
	csv, err := base.GetBase()
	if err != nil {
		return fmt.Errorf("error getting ClusterServiceVersion base: %v", err)
	}
	if err = writeCSVFile(csv, basePath); err != nil {
		return fmt.Errorf("error write CSV base: %v", err)
	}
}
```

Then in tests:

```go
func csvWriteFuncFor(expCSV *v1alpha1.ClusterServiceVersion, expPath string) csvWriteFunc {
	return func(csv *v1alpha1.ClusterServiceVersion, path string) error {
		ExpectWithOffset(3, csv).To(BeEquivalentTo(expCSV))
		ExpectWithOffset(3, path).To(Equal(expPath))
		return nil
	}
}

var _ = Describe("run", func() {
	...
	writeCSV = csvWriteFuncFor(someExpCSV, someExpPath)
	Expect(run(cfg)).NotTo(HaveOccurred())
	...
}
```